simulation/traffic_simulator/src/utils/distance.cpp

```diff
@@ -214,6 +214,9 @@ auto distanceToLeftLaneBound(
   const traffic_simulator_msgs::msg::BoundingBox & bounding_box, const lanelet::Ids & lanelet_ids,
   const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils_ptr) -> double
 {
+  if (lanelet_ids.empty()) {
+    THROW_SEMANTIC_ERROR("Failing to calculate distanceToLeftLaneBound given an empty vector.");
+  }
```

Comment:
It seems, you changed the behavior of the functions in traffic_simulator.
So, please run regression test before pull request open.